Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -312,10 +348,15 @@ bool Launcher::open_file_url(const URL& url)
     if ((st.st_mode & S_IFMT) == S_IFREG && st.st_mode & (S_IXUSR | S_IXGRP | S_IXOTH))
         return spawn(url.path(), {});
 
-    auto extension_parts = url.path().to_lowercase().split('.');
-    String extension = {};
-    if (extension_parts.size() > 1)
-        extension = extension_parts.last();
+    auto extension = LexicalPath::extension(url.path()).to_lowercase();
+    auto mime_type = mime_type_for_file(url.path());
+
+    auto handlers = m_file_handlers;
+    auto mime_type_or_extension = extension;
+    if (mime_type.has_value()) {
+        handlers = m_mime_handlers;
+        mime_type_or_extension = mime_type.value();
+    }
```

Comment:
Because `handlers` is a value not a reference, you're copying a HashTable when you initialise it, and copying another HashTable if there's a mime-type. You can avoid that by removing it, and then selecting the correct HashTable below. (See next comment)

```suggestion
    auto mime_type_or_extension = extension;
    if (mime_type.has_value()) {
        mime_type_or_extension = mime_type.value();
    }
```